contest/remote/lib/fetcher.py

```diff
@@ -128,8 +128,13 @@ def _run_once(self):
         self._run_test(to_test)
 
     def run(self):
+        count = int(os.environ.get('NIPA_FETCHER_COUNT', 2**100))
```

Comment:
You mean something like:

```diff
diff --git a/contest/remote/lib/fetcher.py b/contest/remote/lib/fetcher.py
index 9165bea..6c38f3a 100644
--- a/contest/remote/lib/fetcher.py
+++ b/contest/remote/lib/fetcher.py
@@ -14,6 +14,7 @@ class Fetcher:
         self._cb = cb
         self._cbarg = cbarg
         self.name = name
+        self.count = -1

         self._branches_url = branches_url
         self._check_secs = check_sec
@@ -52,6 +53,8 @@ class Fetcher:
         elif first_run == "next":
             # unless there's a crazy race or time error this will skip newest branch
             self._last_date = datetime.datetime.now(datetime.UTC)
+        elif first_run.isdigit():
+            self.count = int(first_run);

     def _result_set(self, branch_name, url):
         try:
@@ -128,13 +131,12 @@ class Fetcher:
         self._run_test(to_test)

     def run(self):
-        count = int(os.environ.get('NIPA_FETCHER_COUNT', 2**100))
         while True:
-            if count <= 0:
+            if self.count == 0:
                 print("Fetcher is done")
                 break
             self._run_once()
-            count -= 1
+            self.count -= 1
             try:
                 time.sleep(self._check_secs)
```

I didn't quite get this:
> Could you add this to the config file (section executor, name... single_shot?

You mean to tdc's config file?

Reply:
Yeah it works, will do that instead